fastplotlib/graphics/line.py

```diff
@@ -83,3 +83,35 @@ def __init__(
 
         if z_position is not None:
             self.world_object.position.z = z_position
+
+    def _set_feature(self, feature: str, new_data: Any, indices: Any = None):
+        if not hasattr(self, "_previous_data"):
+            self._previous_data = {}
+        elif hasattr(self, "_previous_data"):
+            self._reset_feature(feature)
+        # if feature in self._feature_events:
+        feature_instance = getattr(self, feature)
+        if indices is not None:
+            previous = feature_instance[indices].copy()
+            feature_instance[indices] = new_data
+        else:
+            previous = feature_instance[:].copy()
+            feature_instance[:] = new_data
```

Comment:
similar, this should be like this which works for both indexable and non-indexable features. 

```python
feature_instance._set(new_data)
```